single_kernel_mongo/status.py

```diff
@@ -91,17 +93,26 @@ def clear_status(self, status: StatusBase) -> None:
     def get_statuses(self) -> Statuses:
         """Collects the statuses of all managers."""
         if self.operator.name == CharmKind.MONGOD:
+            # Getting first status is a temporary work around for using our current get_statuses
+            #  functions. When advacned statuses are implemented they will prioritised + shown.
+            mongo_statuses = self.operator.mongo_manager.get_statuses()
+            shard_statuses = self.operator.shard_manager.get_statuses()
+            config_server_statuses = self.operator.config_server_manager.get_statuses()
+            pbm_statuses = self.operator.backup_manager.get_statuses()
+            ldap_status = self.operator.ldap_manager.get_statuses()
             return Statuses(
-                mongodb=self.operator.mongo_manager.get_status(),
-                shard=self.operator.shard_manager.get_status(),
-                config_server=self.operator.config_server_manager.get_status(),
-                pbm=self.operator.backup_manager.get_status(),
-                ldap=self.operator.ldap_manager.get_status(),
+                mongodb=next(iter(mongo_statuses), ActiveStatus()),
+                shard=next(iter(shard_statuses), None),
+                config_server=next(iter(config_server_statuses), None),
+                pbm=next(iter(pbm_statuses), None),
+                ldap=next(iter(ldap_status), None),
             )
         # Mongos case
+        mongos_statuses = self.operator.get_statuses()
+        ldap_status = self.operator.ldap_manager.get_statuses()
         return Statuses(
-            mongodb=self.operator.get_sanity_check_status() or ActiveStatus(),
-            ldap=self.operator.ldap_manager.get_status(),
+            mongodb=next(iter(mongos_statuses), ActiveStatus()),
+            ldap=next(iter(ldap_status), None),
```

Comment:
What is the difference between `None` and `ActiveStatus` ?

Reply:
IIRC mongodb should always return at least an ActiveStatus but the others can return None (aka nothing to report).
However please note that this file will disappear in the next PR as we'll use the library to handle statuses.